src/xcode/ENA/ENA/Source/Services/HealthCertificateValidation/__tests__/HealthCertificateValidationServiceValidationTests.swift

```diff
@@ -1448,6 +1450,89 @@ class HealthCertificateValidationServiceValidationTests: XCTestCase {
 		// THEN
 		XCTAssertEqual(mappedTime, expectedDate)
 	}
+
+	// MARK: - CertLogicEngineValidation
+
+	func test_CertLogicEngineValidation() throws {
+		guard let jsonData = certLogicTestData else {
+			XCTFail("Could not load json data.")
+			return
+		}
+
+		let testData = try JSONDecoder().decode(CertEngineTestData.self, from: jsonData)
+
+		guard let valueSetsData = Data(base64Encoded: testData.general.valueSetProtocolBuffer),
+			  let valueSets = try? SAP_Internal_Dgc_ValueSets(serializedData: valueSetsData) else {
+			XCTFail("Could not load valueSets.")
+			return
+		}
+
+		let valueSetsStub = ValueSetsStub()
+		valueSetsStub.valueSets = valueSets
+
+		let expectation = expectation(description: "Validation should complete for every test case.")
+		expectation.expectedFulfillmentCount = testData.testCases.count
+
+		for testCase in testData.testCases {
+			let mockVerifier = MockVerifier()
+			let mockStore = MockTestStore()
+			let mockClient = ClientMock()
+
+			guard let package = try? makeSAPDownloadedPackage(with: testCase.rules) else {
+				XCTFail("Could not create package.")
+				return
+			}
+
+			mockClient.onGetDCCRules = { _, _, completion in
+				let package = PackageDownloadResponse(
+					package: package,
+					etag: ""
+				)
+				completion(.success(package))
+			}
+
+			let validationService = HealthCertificateValidationService(
+				store: mockStore,
+				client: mockClient,
+				vaccinationValueSetsProvider: valueSetsStub,
+				signatureVerifier: mockVerifier,
+				validationRulesAccess: ValidationRulesAccess()
+			)
+
+			let certificate = try HealthCertificate(base45: testCase.dcc)
+			let country = try XCTUnwrap(Country(countryCode: testCase.countryOfArrival))
+
+			validationService.validate(
+				healthCertificate: certificate,
+				arrivalCountry: country,
+				validationClock: Date(timeIntervalSince1970: TimeInterval(testCase.validationClock))
+			) { result in
+
+				guard case let .success(validationReport) = result else {
+					XCTFail("Success expected for validation result.")
+					return
+				}
+
+				switch validationReport {
+				case .validationFailed(let validatonResults),
+					 .validationOpen(let validatonResults),
+					 .validationPassed(let validatonResults):
+
+					let passCount = validatonResults.filter { $0.result == .passed }.count
+					let openCount = validatonResults.filter { $0.result == .open }.count
+					let failCount = validatonResults.filter { $0.result == .fail }.count
```

Comment:
```suggestion
				case let .validationFailed(validatonResultsFailed),
					 let .validationOpen(validatonResultsOpen),
					 let .validationPassed(validatonResultsPassed):

					let passCount = validatonResultsPassed.filter { $0.result == .passed }.count
					let openCount = validatonResultsOpen.filter { $0.result == .open }.count
					let failCount = validatonResultsFailed.filter { $0.result == .fail }.count
```

Reply:
a little bit confusing when all the properties have the same name

Reply:
They all need to have the same name to have them available in every case

Reply:
They all need to have the same name to have them available in every case